packages/block-library/src/file/edit.native.js

```diff
@@ -326,12 +363,21 @@ export class FileEdit extends Component {
 export default compose( [
 	withDispatch( ( dispatch ) => {
 		const { createInfoNotice } = dispatch( 'core/editor' );
+
 		return {
 			closeSettings() {
 				dispatch( 'core/edit-post' ).closeGeneralSidebar();
 			},
 			createInfoNotice,
 		};
 	} ),
+	withSelect( ( select, props ) => {
+		const { attributes } = props;
+		const { id } = attributes;
+		return {
+			media:
+				id === undefined ? undefined : select( 'core' ).getMedia( id ),
```

Comment:
@etoledom will this `getMedia` call work if the upload wasn't actually done already? I remember seeing a comment about the current implementation being a fake upload hence my question.

Reply:
As we talked internally, is true that we are faking the upload on the demo apps, but on the parent apps is all working as expected, so this feature is best to be tested with the parent apps.

Even though, even with the fake upload, I believe we are sending all needed information back to gutenberg for it to create the appropriate media object, so this is worth checking too 👍

Reply:
> On a second thought, even with the fake upload, I believe we are sending all needed information back to gutenberg for it to create the appropriate media object, so this is worth checking too 👍

Indeed! This is what I was wondering as well. I will take another look. 🙏